runtime/vm/callin.cpp

```diff
@@ -883,6 +883,24 @@ sendFromMethodDescriptorString(J9VMThread *currentThread, J9UTF8 *descriptor, J9
 				currentThread->returnValue2 = (UDATA)J9VMJAVALANGINVOKEMETHODTYPE_FROMMETHODDESCRIPTORSTRINGAPPENDARG_METHOD(vm);
 			}
 			c_cInterpreter(currentThread);
+			if (VM_VMHelpers::exceptionPending(currentThread)) {
```

Comment:
Rather than duplicating this code natively, wouldn't it be cleaner to call `vmResolveFromMethodDescriptorString()` directly and let the java code convert the TNPE to a CNFE?

Reply:
It will be better in terms of avoiding duplicate code. But, `sendFromMethodDescriptorString` calls `MT.fromMethodDescriptorString` and `MT.fromMethodDescriptorStringAppendArg`. So, the definition of `MH.vmResolveFromMethodDescriptorString` will change. 

```
	private static final MethodType vmResolveFromMethodDescriptorString(String methodDescriptor, ClassLoader loader, Class<?> appendArgumentType) throws Throwable {
		try {
			if (null == appendArgumentType) {
				return MethodType.fromMethodDescriptorString(methodDescriptor, loader);
			} else {
				return MethodType.fromMethodDescriptorStringAppendArg(methodDescriptor, loader, appendArgumentType);
			}
		} catch (TypeNotPresentException e) {
			throw throwNoClassDefFoundError(e);
		}
	}
```

Currently, `MT.fromMethodDescriptorStringAppendArg` is private. It will change to a public method since it will be used in `MethodHandle.java`.

MH resolve code path will become slightly longer due to this change. @DanHeidinga do you approve the above changes?

Reply:
It cannot become a public method as that changes the public api for a class in the java.lang.invoke.* namespace.

It can become package or default access which will allow it to be called anywhere in the package without becoming API.

For the MH resolve, this shouldn't be an issue as the resolve occurs once per item.

Reply:
completed.